language/puppet-functions.md

```diff
@@ -7,16 +7,20 @@ Grammar
 ---
 
     FunctionDefinition
-      : 'function' parameters = ParameterList? '{' statements += statements '}'
+      : 'function' name = NAME parameters = ParameterList? ReturnType? '{' statements += statements '}'
```

Comment:
Agree, it does however vary throughout the entire specification suite (names/assignments are only used where needed). IMO, names should only be used when they add something - they otherwise only makes it harder to read